api/konnect/v1alpha1/konnect_cloudgateway_dataplane_configuration_types.go

```diff
@@ -0,0 +1,257 @@
+	//
+	// +kubebuilder:validation:Required
+	NetworkRef NetworkRef `json:"networkRef"`
+
+	// Autoscale configuration for the data-plane group.
+	//
+	// +kubebuilder:validation:Required
+	Autoscale ConfigurationDataPlaneGroupAutoscale `json:"autoscale"`
+
+	// Array of environment variables to set for a data-plane group.
+	//
+	// +kubebuilder:validation:Optional
+	Environment []ConfigurationDataPlaneGroupEnvironmentField `json:"environment,omitempty"`
+}
+
+// ConfigurationDataPlaneGroupEnvironmentField specifies an environment variable field for the data-plane group.
+type ConfigurationDataPlaneGroupEnvironmentField struct {
+	// Name of the environment variable field to set for the data-plane group. Must be prefixed by KONG_.
+	//
+	// +kubebuilder:validation:MinLength=1
+	// +kubebuilder:validation:MaxLength=63
+	// +kubebuilder:validation:Required
+	// +kubebuilder:validation:Pattern="^KONG_."
+	Name string `json:"name"`
+	// Value assigned to the environment variable field for the data-plane group.
+	//
+	// +kubebuilder:validation:MinLength=1
+	// +kubebuilder:validation:MaxLength=63
+	// +kubebuilder:validation:Required
+	Value string `json:"value"`
+}
+
+// ConfigurationDataPlaneGroupAutoscale specifies the autoscale configuration for the data-plane group.
+//
+// +kubebuilder:validation:XValidation:rule="self.type == 'ConfigurationDataPlaneGroupAutoscaleStatic' ? has(self.static) : true",message="static is required when type is ConfigurationDataPlaneGroupAutoscaleStatic"
+// +kubebuilder:validation:XValidation:rule="self.type == 'ConfigurationDataPlaneGroupAutoscaleAutopilot' ? has(self.autopilot) : true",message="autopilot is required when type is ConfigurationDataPlaneGroupAutoscaleAutopilot"
+type ConfigurationDataPlaneGroupAutoscale struct {
+	// ConfigurationDataPlaneGroupAutoscaleStatic specifies the static configuration for the data-plane group.
+	//
+	// +kubebuilder:validation:Optional
+	ConfigurationDataPlaneGroupAutoscaleStatic *ConfigurationDataPlaneGroupAutoscaleStatic `json:"static,omitempty"`
+
+	// ConfigurationDataPlaneGroupAutoscaleAutopilot specifies the autoscale configuration for the data-plane group.
+	//
+	// +kubebuilder:validation:Optional
+	ConfigurationDataPlaneGroupAutoscaleAutopilot *ConfigurationDataPlaneGroupAutoscaleAutopilot `json:"autopilot,omitempty"`
+
+	// Type of autoscaling to use.
+	//
+	// +kubebuilder:validation:Required
+	// +kubebuilder:validation:Enum=ConfigurationDataPlaneGroupAutoscaleStatic;ConfigurationDataPlaneGroupAutoscaleAutopilot
+	Type sdkkonnectcomp.ConfigurationDataPlaneGroupAutoscaleType `json:"type"`
+}
+
+// ConfigurationDataPlaneGroupAutoscaleAutopilot specifies the autoscale configuration for the data-plane group.
+type ConfigurationDataPlaneGroupAutoscaleAutopilot struct {
+	// Base number of requests per second that the deployment target should support.
+	//
+	// +kubebuilder:validation:Required
+	BaseRps int64 `json:"base_rps"`
```

Comment:
Do we know of a default Konnect value that can be set here as well?

Reply:
There's no default for this in the [SDK](https://github.com/Kong/sdk-konnect-go/blob/08223d25420edd301cacdad9a94630a4114cd829/models/components/configurationdataplanegroupautoscaleautopilot.go#L36-L37) or the OpenAPI spec itselfhttps://github.com/Kong/sdk-konnect-go/blob/d511d983a17fb1be7ddd4950cc571f4482990127/openapi.yaml#L10620-L10625 So I'd leave it as is.